pycarol/bigquery.py

```diff
@@ -265,37 +268,49 @@ def query(
             dataset_id: BigQuery dataset ID, if not provided, it will use the default
                         one.
             return_dataframe: Return dataframe if True.
+            return_job_id : If True, returns an tuple containing the query results with the job-id on BigQuery platform.
+            retry: Custom google.api_core.retry.Retry object to adjust Google`s BigQuery API
+                   calls, to custom timeout and exceptions to retry.
 
         Returns:
             Query result.
 
         Usage:
-
         .. code:: python
 
 
             from pycarol import BQ, Carol
 
             bq = BQ(Carol())
             query = 'select * from invoice limit 10'
+            
+            #Normal use
             df = bq.query(query, return_dataframe=True)
+
+            #Custom retry object
+            from google.api_core.retry import Retry
+            df = bq.query(query, return_dataframe=True, retry=Retry(initial=2, multiplier=2, maximum=60, timeout=200))
+
+            #Getting BigQuery`s Job-id (Util for debugging in platform)
+            df, job_id_string = bq.query(query, return_dataframe=True, return_job_id=True)
+
         """
         service_account = self._token_manager.get_token().service_account
         client = self._generate_client(service_account)
 
         dataset_id = dataset_id or self._dataset_id
         job_config = bigquery.QueryJobConfig(default_dataset=dataset_id)
-        results_job = client.query(query, job_config=job_config)
+        results_job = client.query(query, retry=retry, job_config=job_config) if retry else client.query(query, job_config=job_config)
 
         results = [dict(row) for row in results_job]
 
         if return_dataframe is False:
-            return results
+            return results if not return_job_id else (results, results_job.job_id)
```

Comment:
If you are only interested in jobID, you could return before the line 305, you dont need to load the full dataset in memory to return just the jobId.

Reply:
i`m interested in query results and the job id

Reply:
aa, Now Is see it is a tuple with both.